File: vw_c/src/Pipeline/PipelineLayout.cpp
#include "Pipeline/PipelineLayout.h"

#include <VulkanWrapper/Pipeline/PipelineLayout.h>

vw::PipelineLayout *vw_create_pipeline_layout(const vw::Device *device) {
    return new vw::PipelineLayout(vw::PipelineLayoutBuilder(*device).build());
}

void vw_destroy_pipeline_layout(vw::PipelineLayout *pipelineLayout) {
    delete pipelineLayout;
}

File: vw_py/Pipeline/PipelineLayout.py
import bindings_vw_py as bindings
from weakref import finalize

class PipelineLayout:
    def __init__(self, device, pipeline_layout):
        self.device = device
        self.pipeline_layout = pipeline_layout
        finalize(self, bindings.vw_destroy_pipeline_layout, pipeline_layout)

class PipelineLayoutBuilder:
    def __init__(self, device):
        self.device = device

    def build(self):
        pipeline_layout = bindings.vw_create_pipeline_layout(self.device.device)
        return PipelineLayout(self.device, pipeline_layout)